#!/usr/bin/env python
PACKAGE = "spc_uav_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("c_mass", 	double_t, 0, "Admittence filter mass",                    0.9, 0.00,  5.00)
gen.add("c_J_xx", 	double_t, 0, "Admittence filter Inertia Matrix xx", 	  0.9, 0.00,  1.00)
gen.add("c_J_yy", 	double_t, 0, "Admittence filter Inertia Matrix yy", 	  0.9, 0.00,  1.00)
gen.add("c_J_zz", 	double_t, 0, "Admittence filter Inertia Matrix zz", 	  0.9, 0.00,  1.00)
gen.add("c_KD_r_x", double_t, 0, "Admittence filter damper rot gain x", 	  0.5, 0.0,  10.0)
gen.add("c_KD_r_y", double_t, 0, "Admittence filter damper rot gain y", 	  0.5, 0.0,  10.0)
gen.add("c_KD_r_z", double_t, 0, "Admittence filter damper rot gain z", 	  0.5, 0.0,  10.0)
gen.add("c_KD_t_x", double_t, 0, "Admittence filter damper trans gain x", 	  0.5, 0.0,  10.0)
gen.add("c_KD_t_y", double_t, 0, "Admittence filter damper trans gain y", 	  0.5, 0.0,  10.0)
gen.add("c_KD_t_z", double_t, 0, "Admittence filter damper trans gain z", 	  0.5, 0.0,  10.0)
gen.add("c_KP_r_xx",double_t, 0, "Admittence filter Spring K_Pr Matrix xx",   3.0, 0.0,  10.0)
gen.add("c_KP_r_yy",double_t, 0, "Admittence filter Spring K_Pr Matrix yy",   3.0, 0.0,  10.0)
gen.add("c_KP_r_zz",double_t, 0, "Admittence filter Spring K_Pr Matrix zz",   3.0, 0.0,  10.0)
gen.add("c_KP_t_xx",double_t, 0, "Admittence filter Spring K_Pt Matrix xx",   3.0, 0.0,  10.0)
gen.add("c_KP_t_yy",double_t, 0, "Admittence filter Spring K_Pt Matrix yy",   3.0, 0.0,  10.0)
gen.add("c_KP_t_zz",double_t, 0, "Admittence filter Spring K_Pt Matrix zz",   3.0, 0.0,  10.0)
exit(gen.generate(PACKAGE, "spc_uav_control", "Admittance_Control_Gains"))
